coreplugins/cesiumion/public/TaskView.jsx

```diff
@@ -0,0 +1,287 @@
+		if (items.some(item => item.isTask && !item.isError)) {
+			const timeout = 4000 / (isTasksDialog ? 2 : 1);
+			this.timeoutHandler = setTimeout(this.refreshAssets, timeout);
+		}
+	};
+
+	onCleanStatus = ({ updated = false }) => {
+		if (!updated || this.refreshAssets == null) return;
+		this.refreshAssets();
+	};
+
+	onErrorUploadDialog = msg => {
+		this.setState({ error: msg });
+		this.onHideUploadDialog();
+	};
+
+	handleAssetSelect = data => asset => {
+		const idMap = data.items
+			.filter(item => item.isExported)
+			.reduce((accum, item) => {
+				accum[item.type] = item.id;
+				return accum;
+			}, {});
+		window.open(`https://cesium.com/ion/assets/${idMap[asset]}`);
+	};
+
+	componentWillUnmount() {
+		if (this.timeoutHandler !== null) {
+			clearTimeout(this.timeoutHandler);
+			this.timeoutHandler = null;
+		}
+		if (this.cancelableFetch !== null) {
+			try {
+				this.cancelableFetch.cancel();
+				this.cancelableFetch = null;
+			} catch (exception) {
+				console.warning("Failed to clear dead requests");
+				console.warning(exception);
+			}
+		}
+
+		this.refreshAssets = null;
+	}
+
+	render() {
+		const { task, token } = this.props;
+		const {
+			isTasksDialog,
+			isUploadDialogLoading,
+			isRefreshTask,
+			currentAsset
+		} = this.state;
+		const isUploadDialog = currentAsset !== null;
+		const assetName = isUploadDialog ? AssetStyles[currentAsset].name : "";
+
+		return (
+			<AppContext.Provider value={this.props}>
+				<ErrorMessage bind={[this, "error"]} />
+				<div className={"ion-dropdowns"}>
+					<TaskFetcher
+						path={"share"}
+						onLoad={this.onAssetsRefreshed}
+						onBindRefresh={method => (this.refreshAssets = method)}
+					>
+						{({ isError, data = {} }) => {
+							// Asset Export and View Selector
+							const { items = [] } = data;
+							const available = items
+								.filter(
+									item =>
+										(!item.isExported && !item.isTask) ||
+										item.isError
+								)
+								.map(item => item.type);
+							const exported = items
+								.filter(item => item.isExported)
+								.map(item => item.type);
+
+							// Tasks Selector
+							const processing = items.filter(
+								item => item.isTask
+							);
+							const isTasks = processing.length > 0;
+							const isErrors = processing.some(
+								item => item.isError
+							);
+
+							return (
+								<Fragment>
+									{available.length > 0 && (
+										<IonAssetButton
+											assets={available}
+											onSelect={this.onOpenUploadDialog}
+										>
+											Tile in Cesium Ion
+										</IonAssetButton>
+									)}
+
+									{exported.length > 0 && (
+										<IonAssetButton
+											assets={exported}
+											onSelect={this.handleAssetSelect(
+												data
+											)}
+										>
+											View in Cesium Ion
+										</IonAssetButton>
+									)}
+									{items.length <= 0 && (
+										<button
+											className={"ion-btn btn btn-primary btn-sm"}
```

Comment:
Button component with styles or nah? Noticed we duplicated the classes but it's fine if the duplication isn't that bad

Reply:
This was the easiest and fastest way I refactored to fix the glitches at the beginning. Not sure if there's duplication but it works.